addon/components/polaris-popover.js

```diff
@@ -90,11 +93,73 @@ export default Component.extend({
   /*
    * Internal properties.
    */
+  verticalPosition: computed('preferredPosition', function() {
+    let preferredPosition = this.get('preferredPosition');
+
+    if (preferredPosition === 'above') {
+      return preferredPosition;
+    }
+
+    if (preferredPosition === 'below') {
+      return preferredPosition;
+    }
+
+    // If set to `mostSpace`, the value will be calculated
+    // and set when the user opens the popover.
+
+    return null;
+  }),
```

Comment:
`.readOnly()`?

Reply:
This gets `set` by the `onOpen` action if the position is set to `"mostSpace"`, so it can't be readOnly

Reply:
Hmm... seems like an _unlikely_ scenario, but `set`ting this in `onOpen` will mean that if the host app passes in a dynamic `preferredPosition` and that value changes from 'mostSpace' after first open, the popover won't honour that change because `onOpen` will overwrite the CP behaviour here. You _could_ add an explicit setter to this CP, and that way I think that (really tiny) edge case would be avoided...

Reply:
`onOpen` doesn't overwrite anything. `verticalPosition` is still a CP, so if `preferredPosition` dynamically changes, it will re-set `verticalPosition` to the new value, and then when `onOpen` runs it will _not_ compute a new value for `verticalPosition` because it only computes a new value if `preferredPosition` is set to `"mostSpace"`.

Reply:
After `set`ting `verticalPosition` on [line 161](https://github.com/smile-io/ember-polaris/blob/3915ec85eb771e0441db38b6de7f8c5d8762c9c3/addon/components/polaris-popover.js#L161), my understanding is that `verticalPosition` will _no longer be a CP_ - the `set` will replace the entire CP with the literal value that you're setting it to, unless the CP has its own `set` defined as well. Think of it like this, where `someObject` is the result of the `computed` call:

```js
  ... other props...

  myComputed: someObject,

  ... other stuff...
```

If you then `this.set('myComputed', 'something else')`, you'll _replace `someObject` with the new string literal_. The CP "contained" in `someObject` will be torn down, and will no longer be computed on dependency changes. From then on, `this.get('myComputed')` will _always_ return `'something else'`.

You can work around this by defining the CP as

```js
myComputed: computed('dep1', 'dep2', {
  get() {
    return 'whatever';
  },

  set(key, value) {
    return value;
  }
),
```

The `set` part ensures that the CP remains after a `this.set('myComputed', 'something else')`, and the value will still update after `dep1` or `dep2` change...